wolfcrypt/src/ed25519.c

```diff
@@ -386,18 +386,75 @@ int wc_ed25519ph_sign_msg(const byte* in, word32 inLen, byte* out,
 
 #ifdef HAVE_ED25519_VERIFY
 
+/*
+   sig        is array of bytes containing the signature
+   sigLen     is the length of sig byte array
+   key        Ed25519 public key
+   return     0 on success
+   type       variant to use -- Ed25519, Ed25519ctx, or Ed25519ph
+   context    extra signing data
+   contextLen length of extra signing data
+*/
+int wc_ed25519_verify_msg_init(const byte* sig, word32 sigLen, ed25519_key* key,
+                               byte type, const byte* context, byte contextLen)
+{
+    int    ret;
+
+    /* sanity check on arguments */
+    if (sig == NULL || key == NULL ||
+        (context == NULL && contextLen != 0)) {
+        return BAD_FUNC_ARG;
+    }
+
+    /* check on basics needed to verify signature */
+    if (sigLen != ED25519_SIG_SIZE || (sig[ED25519_SIG_SIZE-1] & 224))
+        return BAD_FUNC_ARG;
+
+    /* find H(R,A,M) and store it as h */
+
+    ret = wc_InitSha512_ex(&key->sha, key->heap, INVALID_DEVID);
+    if (ret != 0)
+        return ret;
+    if (type == Ed25519ctx || type == Ed25519ph) {
+        ret = wc_Sha512Update(&key->sha, ed25519Ctx, ED25519CTX_SIZE);
+        if (ret == 0)
+            ret = wc_Sha512Update(&key->sha, &type, sizeof(type));
+        if (ret == 0)
+            ret = wc_Sha512Update(&key->sha, &contextLen, sizeof(contextLen));
+        if (ret == 0 && context != NULL)
+            ret = wc_Sha512Update(&key->sha, context, contextLen);
+    }
+    if (ret == 0)
+        ret = wc_Sha512Update(&key->sha, sig, ED25519_SIG_SIZE/2);
+    if (ret == 0)
+        ret = wc_Sha512Update(&key->sha, key->p, ED25519_PUB_KEY_SIZE);
+
+    return ret;
+}
+
+/*
+   msgSegment     an array of bytes containing a message segment
+   msgSegmentLen  length of msgSegment
+   key            Ed25519 public key
+   return         0 on success
+*/
+int wc_ed25519_verify_msg_feed(const byte* msgSegment, word32 msgSegmentLen,
```

Comment:
Agree with Sean this should be named `wc_ed25519_verify_msg_update`.

Reply:
done